### ISSUE TITLE
[$250] [Search v2.1] [App] Create Tag filter

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#46040]
Implement [this section](https://docs.google.com/document/d/1bRV1nPz8SaI_6DHqfehmLnADxv40us_q0JNvmZ1QyaY/edit#bookmark=id.ekk0kmv0drso) of the design doc for the Tag filter only!

cc @adamgrzybowski @Kicu @WojtekBoman @Guccio163

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016fae6113cf2772e6</li>
        <li>Upwork Job ID: 1818046679647611930</li>
        <li>Last Price Increase: 2024-07-29</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Reviewer | 103413091</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @SzymczakJ</details>

**Comment by Kicu** [comment]
Hey I would like to work on this one.
I've already finished `task` and `status` filters, Im just waiting on another PR to be merged.

**Comment by Kicu** [comment]
Actually feels to me like this filter might be very similar to the Category one, so I will until this is finalized: https://github.com/Expensify/App/pull/46197

And then just reuse the code.

**Comment by luacmartins** [comment]
Sounds good!

**Comment by ikevin127** [comment]
Not overdue, callback to https://github.com/Expensify/App/issues/46040#issuecomment-2257552649.

**Comment by Kicu** [comment]
@SzymczakJ is taking over for this one, while Im working on From/To

**Comment by luacmartins** [comment]
cool, @SzymczakJ can you please comment here so I can assign this issue to you?

**Comment by SzymczakJ** [comment]
I'm taking this one!

**Comment by SzymczakJ** [comment]
I need to refactor one thing to unify filters(currency, category, expense type, tag, tax) this PR should come out of draft tommorow!

**Comment by luacmartins** [comment]
Payment will be handled via https://github.com/Expensify/App/issues/46041

**Comment by luacmartins** [comment]
I'm gonna close this issue since the PR is merged and payment will be handled in https://github.com/Expensify/App/issues/46041
